src/backend/protocol.ts

```diff
@@ -31,12 +33,16 @@ export async function handleProtocol(window: BrowserWindow, args: string[]) {
   }
 
   if (command === 'launch') {
-    let game = getGame(arg, 'legendary').getGameInfo()
+    const runners: Runner[] = ['legendary', 'gog', 'sideload']
 
-    // getGameInfo returns {} so we need to check for underlying value
-    if (!game?.app_name) {
-      game = getGame(arg, 'gog').getGameInfo()
-    }
+    const game = runners
+      .map((runner) =>
+        runner === 'sideload'
+          ? getAppInfo(arg)
+          : getGame(arg, runner).getGameInfo()
+      )
+      .filter(({ app_name }) => app_name)
+      .shift()
 
     if (!game?.app_name) {
```

Comment:
This could now be simplified to just
```suggestion
    if (!game) {
```